src/classes/RD2_EnablementService.cls

```diff
@@ -104,4 +104,40 @@ public with sharing class RD2_EnablementService {
         return 
             UTIL_CustomSettingsFacade.getRecurringDonationsSettings().isRecurringDonations2Enabled__c == true;
     }
+
+    /**
+     * @description Returns true if RecurringType__c 'Fixed' picklist option available
+     */
+    @TestVisible
+    public static Boolean fixedOptionAvailable {
+        get {
+            if (fixedOptionAvailable == null) {
+                fixedOptionAvailable =
+                    UTIL_Describe.isValidPicklistOption(String.valueOf(npe03__Recurring_Donation__c.SObjectType),
+                        SObjectType.npe03__Recurring_Donation__c.fields.RecurringType__c.Name,
+                        RD2_Constants.RECURRING_TYPE_FIXED
+                );
+            }
+            return fixedOptionAvailable;
+        }
+        private set;
+    }
+
+    public static Boolean isFixedLength(npe03__Recurring_Donation__c rd) {
```

Comment:
ApexDoc comment block needed here. Would probably be helpful if the description explained the logic and why it's there.